src/Root.tsx

```diff
@@ -1,14 +1,48 @@
+/* eslint-disable import/no-extraneous-dependencies */
 import { i18n } from '@lingui/core';
 import { I18nProvider } from '@lingui/react';
+import { configureChains } from '@wagmi/core';
+import { mainnet, goerli } from '@wagmi/core/chains';
+import { ConnectKitProvider } from 'connectkit';
 import type { GatsbyBrowser } from 'gatsby';
+import { WagmiConfig, createConfig } from 'wagmi';
+import { MetaMaskConnector } from 'wagmi/connectors/metaMask';
+import { WalletConnectConnector } from 'wagmi/connectors/walletConnect';
+import { infuraProvider } from 'wagmi/providers/infura';
+import { publicProvider } from 'wagmi/providers/public';
 
 import { Layout, SnapsProvider } from './components';
 import { messages } from './locales/en/messages';
 import { createStore } from './store';
 
+import { getConfig } from '@/utils/config/config';
+import { lineaMainnet, lineaTestnet } from '@/utils/customChains';
+
 // eslint-disable-next-line import/no-unassigned-import, import/extensions
 import './assets/fonts/fonts.css';
 
+/* Wagmi */
+const config = getConfig();
+
+const { chains, publicClient } = configureChains(
+  [mainnet, goerli, lineaMainnet, lineaTestnet],
+  [infuraProvider({ apiKey: config.infuraId }), publicProvider()],
+);
+
+const wagmiConfig = createConfig({
```

Comment:
Can you explain why we need all of these dependencies? What do they allow that we can't do otherwise etc.?

Reply:
Using `connectkit` implies to installing ` wagmi` and `viem` as well. 
Following [doc](https://docs.family.co/connectkit/getting-started#getting-started-section-2-api-keys),  `ConnectKit` requires an infura  or  alchemy  API key depending on which networks your dApp requires.

`[ConfigureChains](https://wagmi.sh/react/providers/configuring-chains)`  allows you to configure your chains with RPC Providers. I added our used chains within Metamask (goerli, linea, etc).